### PR TITLE
fix_: cli not logging all messages

### PR DESCRIPTION
**Pull request by plopezlpz**
CLI is not logging some messages received, because using `messenger.RetrieveAll()` will not report all messages if called from two different routines

relates: #5266

### CODE REVIEW

File: cmd/status-cli/serve.go [hunk 1]
```diff
@@ -44,14 +40,32 @@ func serve(cCtx *cli.Context) error {
...
 
-	// Retrieve for messages
-	var wg sync.WaitGroup
-	msgCh := make(chan string)
+	// Using the mobile signal handler to listen for received messages
+	// because if we call messenger.RetrieveAll() from different routines we will miss messages in one of them
+	// and the retrieve messages loop is started when starting a node, so we needed a different appproach,
+	// alternatively we could have implemented another notification mechanism in the messenger, but this signal is already in place
+	msignal.SetMobileSignalHandler(msignal.MobileSignalHandler(func(s []byte) {
+		if strings.Contains(string(s), `"type":"messages.new"`) {
```
Comment (kaichaosun) [inline cmd/status-cli/serve.go]: This looks error prone, can we unmarshal first and make the check after unmarshal, maybe the type field?

File: cmd/status-cli/serve.go [hunk 2]
```diff
@@ -60,22 +74,29 @@ func serve(cCtx *cli.Context) error {
...
+		signal.Notify(sig, syscall.SIGINT, syscall.SIGTERM)
+		<-sig
+		cancel()
 	}()
 
 	// Send message if mutual contact exists
-	sem := make(chan struct{}, 1)
-	wg.Add(1)
-	go cli.sendMessageLoop(ctx, SendInterval, &wg, sem, cancel)
+	cli.sendMessageLoop2(ctx)
```
Comment (kaichaosun) [inline cmd/status-cli/serve.go]: Do we still need to keep the `sendMessageLoop`, if yes, better refactor the method name to show the difference.

File: cmd/status-cli/serve.go (hunk 1 above)
Comment (plopezlpz) [reply cmd/status-cli/serve.go]: yes, thanks

File: cmd/status-cli/serve.go (hunk 2 above)
Comment (plopezlpz) [reply cmd/status-cli/serve.go]: refactored, thanks

**Review by kaichaosun — APPROVED** [review]

**Pull request merged**